Load uswds with webpack

jseppi opened the pull request:
As mentioned in [this comment](https://github.com/18F/federalist/pull/949#issuecomment-305523075), USWDS styles/assets are currently managed by a series of npm tasks (listed in `package.json`). One problem that currently exists with that method is that the paths to assets (like images referenced in the uswds.css) are not correct.

This PR brings uswds into our webpack build, which:
- solves the asset paths problem (including getting rid of custom style rules to assets that were overridden)
- adds all the uswds style rules into our main styles bundle, which reduces the number of client requests
- removes the 4 uswds npm tasks, thus simplifying further our `package.json` scripts
- fixes a 404 on an asset on the `site/X/settings` view. This is related to #955.

To Do:
- [x] figure out and fix the failing `mocha` run. Potential fix in this commit: https://github.com/coryhouse/react-slingshot/commit/16ec28c9029bf7e2b65b26c22a1c2daadab427a2
- [x] fix the styling of the delete site button on the `site/X/settings` view (see screenshot below)
- [x] get the npm watch tasks to work. Some options for this:
    - *I went with this*: reload the server on asset change, because it needs to reload `webpack-manifest.json` to get the newest asset names. This will maybe slow down development since the server has to reload.
    - ~only use hashed names in production (by checking `NODE_ENV`). This might be the most "correct" thing to do, but will require changing builds a little bit.~
    - ~read `webpack-manifest.json` on each client request to the main controller. This is the easiest method.~

Now that the asset paths are fixed, I actually found an ugly design bug on `site/X/settings` view:

<img width="461" alt="screen shot 2017-06-01 at 1 30 33 pm" src="https://cloud.githubusercontent.com/assets/697848/26695679/4fa02a26-46d1-11e7-85cd-d26955eb6256.png">

I fixed it by removing the background-image and adding a small margin to the delete button:
<img width="498" alt="screen shot 2017-06-01 at 2 02 44 pm" src="https://cloud.githubusercontent.com/assets/697848/26696184/1fb97cf2-46d3-11e7-8211-caa414b20615.png">

jseppi commented:
@wslack fixing the uswds image paths causes some slight visual differences on the site settings view: basically the alert icons work now. Is this ok?

In the before case, as in production currently, there are `404` errors logged for the missing icons. Whatever you decide, we should at least fix the 404 errors.

Before:

<img width="479" alt="screen shot 2017-06-01 at 2 06 04 pm" src="https://cloud.githubusercontent.com/assets/697848/26696310/9c5778cc-46d3-11e7-9254-e5fc1b7d7b88.png">

After:

<img width="495" alt="screen shot 2017-06-01 at 2 05 37 pm" src="https://cloud.githubusercontent.com/assets/697848/26696315/a2764954-46d3-11e7-859f-0d431aac1e5e.png">

jseppi commented:
@brandocalrissian ready to review! Or, if you prefer to just do one review, I can merge this into my other PR #949

wslack commented:
@jseppi no issues with the missing icons

jseppi commented:
To be clear, you mean you want the icons to not be there, correct? If so the left padding on those boxes should also be reduced in my opinion.

wslack commented:
We should fix the bugs but having the icons isn't a big problem to me; they don't take away much. However, if easier to remove the icons, let's just do that!

jseppi commented:
It's easier to leave them there now that the paths are fixed :) Happy to modify the styling to remove them though.

brandocalrissian commented:
I changed the base branch-- already merged the `939-webpack` branch to staging.

brandocalrissian commented on frontend/sass/styles.scss:
```diff
@@ -1,3 +1,5 @@
+@import '../../node_modules/uswds/dist/css/uswds.css';
```
having the hardcoded path to our project root `node_modules` seems clunky, but I don't see an easy way to avoid it if we want to bring the uswds into sass-land.

jseppi replied:
Yeah, I don't know of a better way. If for some reason the path changes in a future release, the webpack build will fail, so it should be "safe" to use this.

brandocalrissian commented on frontend/components/disclaimer.js:
```diff
@@ -1,13 +1,15 @@
           className="usa-flag_icon"
           alt="US flag signifying that this is a United States Federal Government website"
-          src="/images/uswds/us_flag_small.png"
+          src={flagUrl}
```
Is this related to the travis-ci fail? I was expecting it to be from the change of base branch, but then I saw this:

```
/home/travis/build/18F/federalist/node_modules/uswds/dist/img/us_flag_small.png:1
(function (exports, require, module, __filename, __dirname) { �PNG
```

jseppi replied:
I believe the failure was due to the lint task failing

jseppi replied:
oh and also that invalid character...

jseppi replied:
Hrm, it's apparently something with mocha not an actual problem. Trying to figure out how to make mocha not fail on it...

jseppi commented:
Fixed the mocha problem, @brandocalrissian. The issue was that I am using some webpack features for loading non-js assets via `require(...)`, but mocha of course doesn't know how to handle that in the same way. I've added a new file, `_mochaSetup.js`, that will be run with tests and will override `require(...)` for specific file extensions with a noop function.

jseppi commented:
Since @jmhooper is around this week, I think he could review instead/also.
